File: android/app/src/main/kotlin/games/yandex/wrap/catalog/dtos/FeedBlockDto.kt
package games.yandex.wrap.catalog.dtos

import games.yandex.wrap.catalog.models.FeedBlock
import kotlinx.serialization.Serializable

@Serializable
data class FeedBlockDto(
    val type: String,
    val size: String? = null,
    val title: String = "",
    val items: List<GameDto> = emptyList(),
) {
    fun toDomain(): FeedBlock = FeedBlock(
        type = type,
        size = size,
        title = title,
        items = items.map(GameDto::toDomain),
    )
}

File: ios/UGames/Catalog/DTOs/FeedBlockDTO.swift
import Foundation

struct FeedBlockDTO: Decodable {
    let type: String
    let size: String?
    let title: String?
    let items: [GameDTO]?

    var domain: FeedBlock {
        FeedBlock(type: type, size: size, title: title ?? "", items: items.orEmpty.map { $0.domain })
    }
}
